arangod/V8Server/FoxxQueuesFeature.h

```diff
@@ -43,9 +50,15 @@ class FoxxQueuesFeature final : public application_features::ApplicationFeature
     return _pollInterval;
   }
 
+  // Caller can get this lock
+  // Caller needs to make sure to release this lock after use.
+  arangodb::basics::ReadWriteLock& fileSystemLock();
+
  private:
   double _pollInterval;
   bool _enabled;
+
+  arangodb::basics::ReadWriteLock _fileSystemLock;
```

Comment:
As the lock is acquired only in write mode, you could also turn this into a mutex.